RIOT-OS/RIOT
# sys/psa_crypto: Implement persistent key storage

### Contribution description
This is an implementation of persistent key storage in PSA Crypto. It uses VFS with littlefs2 and MTD.
PSA keys are encoded in CBOR, written to files and stored in flash (or emulated flash, depending on the MTD implementation).

So far this works on `native` and the `nRF52840dk` and it requires that the board supports MTD. This is why it is optional and must be enabled explicitly when building PSA Crypto.

### Testing procedure
`tests/sys/psa_crypto_persistent_storage` and `tests/sys/psa_crypto_cbor_encoder` should pass successfully on the supported platforms.
`examples/psa_crypto` should still build and run without problems.

### Issues/PRs references
Probably needs to be updated once  #19992 is merged, since this does not yet include the module separation.
